plugins/govppmux/govpp_channel.go

```diff
@@ -0,0 +1,122 @@
+package govppmux
+
+import (
+	"time"
+
+	govppapi "git.fd.io/govpp.git/api"
+	"git.fd.io/govpp.git/core"
+	"github.com/ligato/cn-infra/logging/logrus"
+)
+
+const defaultRetryRequestTimeout = 500 * time.Millisecond
+
+// goVppChan implements govpp channel interface. Instance is returned by NewAPIChannel() or NewAPIChannelBuffered(),
+// and contains *govpp.channel dynamic type (vppChan field). Implemented methods allow custom handling of low-level
+// govpp.
+type goVppChan struct {
+	vppChan govppapi.Channel
+	// Retry data
+	retry *retryConfig
+}
+
+// govppRequestCtx is custom govpp RequestCtx.
+type govppRequestCtx struct {
+	// Original request context
+	requestCtx govppapi.RequestCtx
+	// Function allowing to re-send request in case it's granted by the config file
+	sendRequest func(govppapi.Message) govppapi.RequestCtx
+	// Parameter for sendRequest
+	requestMsg govppapi.Message
+	// Retry data
+	retry *retryConfig
+}
+
+// helper struct holding info about retry configuration
+type retryConfig struct {
+	attempts int
+	timeout  time.Duration
+}
+
+// ReceiveReply handles request and returns error if occurred. Also does retry if this option is available.
+func (r *govppRequestCtx) ReceiveReply(reply govppapi.Message) error {
+	var err error
+	// Receive reply from original send
+	if err = r.requestCtx.ReceiveReply(reply); err != nil && err == core.ErrNotConnected {
+		if r.retry != nil && r.retry.attempts > 0 {
+			// Set default timeout between retries if not set
+			if r.retry.timeout == 0 {
+				r.retry.timeout = defaultRetryRequestTimeout
+			}
+			// Try to re-sent requests
+			for i := 1; i <= r.retry.attempts; i++ {
+				logrus.DefaultLogger().Warnf("Retrying message %v: %d", r.requestMsg.GetMessageName(), i)
+				ctx := r.sendRequest(r.requestMsg)
+				if err = ctx.ReceiveReply(reply); err == nil {
+					return nil
+				}
+				time.Sleep(r.retry.timeout)
```

Comment:
As I see it, this is going to sleep even after last failed reply even though just right after the sleep it will break loop (after last attempt) and just then return last error.